aviatrix/resource_aviatrix_spoke_gateway.go

```diff
@@ -370,6 +385,18 @@ func resourceAviatrixSpokeGatewayCreate(d *schema.ResourceData, meta interface{}
 		gateway.EncVolume = "no"
 	}
 
+	enableMonitorSubnets := d.Get("enable_monitor_gateway_subnets").(bool)
+	var excludedInstances []string
+	for _, v := range d.Get("monitor_exclude_list").(*schema.Set).List() {
+		excludedInstances = append(excludedInstances, v.(string))
+	}
+	if enableMonitorSubnets && gateway.CloudType != goaviatrix.AWS && gateway.CloudType != goaviatrix.AWSGOV {
+		return fmt.Errorf("'enable_monitor_gateway_subnets' is only valid for cloud_type = 1 (AWS) or 256 (AWSGOV)")
+	}
+	if !enableMonitorSubnets && len(excludedInstances) != 0 {
+		return fmt.Errorf("'monitor_exclude_list' must be empty if 'enable_monitor_gateway_subnets' is false")
+	}
```

Comment:
Will the list be used if 'enable_monitor_gateway_subnets' is false? If not, maybe the error message is too strict?

Reply:
The issue is that the next plan would give a diff because when `enable_monitor_gateway_subnets = false` then `monitor_exclude_list` will always read as empty. So, I think "failing fast" by giving the error message here is better than waiting for the user to notice they input an invalid config the next time the do a plan.

In general, my opinion is that we should let the user know they have an invalid config as soon as possible.